src/QMCApp/QMCMain.cpp

```diff
@@ -242,87 +242,89 @@ bool QMCMain::execute()
   particle_set_pool_->get(app_log());
   ham_pool_->get(app_log());
   OHMMS::Controller->barrier();
-  if (qmc_common.dryrun)
-  {
-    app_log() << "  dryrun == 1 Ignore qmc/loop elements " << std::endl;
-    myComm->barrier_and_abort("QMCMain::execute");
-  }
   t3.stop();
-  Timer t1;
-  qmc_common.qmc_counter = 0;
-  for (int qa = 0; qa < qmc_action_.size(); qa++)
+  if (qmc_common.dryrun)
   {
-    if (run_time_manager.isStopNeeded())
-      break;
-    xmlNodePtr cur = qmc_action_[qa].first;
-    std::string cname((const char*)cur->name);
-    if (cname == "qmc" || cname == "optimize")
-    {
-      executeQMCSection(cur);
-      qmc_common.qmc_counter++; // increase the counter
-    }
-    else if (cname == "loop")
-    {
-      qmc_common.qmc_counter = 0;
-      executeLoop(cur);
-      qmc_common.qmc_counter = 0;
-    }
-    else if (cname == "cmc")
-    {
-      executeCMCSection(cur);
-    }
-    else if (cname == "debug")
-    {
-      executeDebugSection(cur);
-      app_log() << "  Debug is done. Skip the rest of the input " << std::endl;
-      break;
-    }
+    app_log() << "  dryrun == 1 : Skipping all QMC and loop elements " << std::endl;
```

Comment:
How about making an early return instead of keep a giant scope of 'else'.
I learned this from https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code

Reply:
Sure, it is a matter of style. I prefer the early exit but some people do not.

Reply:
If method is of sane size early exit should not present difficulty reading.